src/metabase/lib/field.cljc

```diff
@@ -77,6 +79,30 @@
     (when (every? some? path)
       (str/join ": " path))))
 
+(defn- clauses-in-stage-by-uuid
+  [query stage-number]
+  (lib.computed/with-cache-none* query [::clauses-in-stage-by-uuid stage-number]
+    (fn []
+      (let [sink (volatile! (transient {}))]
+        (lib.walk/walk-clauses-in-stage
+         (lib.util/query-stage query stage-number)
+         (fn [clause]
+           (when-let [lib-uuid (lib.options/uuid clause)]
+             (vswap! sink assoc! lib-uuid clause))
+           nil))
+        (persistent! @sink)))))
+
+(defn- find-stage-index-and-clause-by-uuid
+  "Find the clause in `query` with the given `lib-uuid`. Return a [stage-index clause] pair, if found."
+  [query stage-number lib-uuid]
+  (let [max-stage (lib.util/canonical-stage-index query stage-number)]
+    (loop [stage-number 0]
```

Comment:
Is there a particular reason why we are not counting down?

Reply:
I feel like there was a reason, but I'm not sure what it was anymore. Some test failure, probably, but perhaps the cause was something else.

I don't think it matters much; it'll process about half the stages on average, I think.